test/support/utils.h

```diff
@@ -1014,26 +1014,39 @@ generate_arithmetic_data(T* input, std::size_t size, std::uint32_t seed)
     }
 }
 
-// Utility that models __estimate_best_start_size in the SYCL backend parallel_for
-// to ensure large enough inputs are used to test the large submitter path.
-// A multiplier to the max n is added to ensure we get a few separate test inputs for
-// this path.
-std::size_t
-get_pattern_for_max_n()
+// Utility that models __estimate_best_start_size in the SYCL backend parallel_for to ensure large enough inputs are
+// used to test the large submitter path. A multiplier to the max n is added to ensure we get a few separate test inputs
+// for this path. For debug testing, only test with a single large n to avoid timeouts. Returns a monotonically increasing
+// sequence for use in testing.
+inline std::vector<std::size_t>
+get_pattern_for_test_sizes()
 {
-#if TEST_DPCPP_BACKEND_PRESENT
+    std::size_t max_n = 0;
+#if TEST_DPCPP_BACKEND_PRESENT && !ONEDPL_FPGA_DEVICE
```

Comment:
What is the purpose of having a separate flow for an FPGA device? Is it because the emulator returns some unusual device properties?

Reply:
We have a separate parallel-for implementation for FPGA: https://github.com/uxlfoundation/oneDPL/blob/dbf9c08048e144e294d29ae8e288d2b0e35667df/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_fpga.h#L56 which does not have a small and large submitter. As a result, we do not need to test these large input sizes here for full coverage, and it is likely very slow if we do.

Reply:
Thanks. Let's leave a small note about the intention...

Reply:
Done